core/routemgmt/activateApi.js

```diff
@@ -0,0 +1,383 @@
+/**
+ *
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Create an API Gateway to action mapping document in database:
+ * https://docs.cloudant.com/document.html#documentCreate
+ *
+ * Parameters (all as fields in the message JSON object)
+ *   host       Required. The database dns host name
+ *   port       Required. The database port number
+ *   protocol   Required. The database protocol (i.e. http, https)
+ *   dbname     Required. The name of the database
+ *   username   Required. The database user name used to access the database
+ *   password   Required. The database user password
+ *   gwUrl      Required. The API Gateway base path (i.e. http://gw.com)
+ *   namespace  Required. The namespace of the API to be activated
+ *   basepath   Required. The basepath of the API to be activated
+ *   docid      Optional. If specified, use this docid to retrieve API doc from DB
+ *
+ * NOTE: The package containing this action will be bound to the following values:
+ *         host, port, protocol, dbname, username, password, gwUrl, gwUser, gwPwd
+ *       As such, the caller to this action should normally avoid explicitly setting
+ *       these values
+ **/
+ var request = require('request');
+ var url = require('url');
+
+function main(message) {
+  var badArgMsg = '';
+  if (badArgMsg = validateArgs(message)) {
+    return whisk.error(badArgMsg);
+  }
+  var dbname = message.dbname;
+
+  var gwInfo = {
+    gwUrl: message.gwUrl,
+  };
+  if (message.gwUser && message.gwPwd) {
+    gwInfo.gwAuth = Buffer.from(message.gwUser+':'+message.gwPwd,'ascii').toString('base64');
+  }
+
+  // Log parameter values
+  console.log('DB host    : '+message.host);
+  console.log('DB port    : '+message.port);
+  console.log('DB protocol: '+message.protocol);
+  console.log('DB username: '+message.username);
+  console.log('DB database: '+message.dbname);
+  console.log('GW URL     : '+message.gwUrl);
+  console.log('GW User    : '+message.gwUser);
+  console.log('GW Pwd     : '+message.gwPwd);
+  console.log('GW Auth    : '+gwInfo.gwAuth);
+  console.log('namespace  : '+message.namespace);
+  console.log('basepath   : '+message.basepath);
+  console.log('docid      : '+message.docid);
+
+  var cloudantOrError = getCloudantAccount(message);
+  if (typeof cloudantOrError !== 'object') {
```

Comment:
Why not return `false` if something went wrong in `getCloudantAccount` instead? Or `null`.